### PR TITLE
Fix `process_protein` to only call the callback once

### PR DESCRIPTION
**Pull request by orourkek**
There was a logic error in `process_protein` that was causing the
callback to be called twice. In a previous patch, the function was
augmented to call `callback(null)` when the protein couldn't be parsed
properly. This is reasonable, but `callback(null)` was being executed
even after a successful `callback(protein)`, because of where the
`callback(null)` was (after the `catch` block.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -193,7 +194,7 @@ var process_protein = function process_protein(protein, callback) {
         } catch (e) {
           console.log(e);
         }
-        return true;
```
Comment (ebeneliason) [inline index.js]: I'm confused by the assertion that the callback was called twice given this return statement, which falls outside the `try/catch` block and should have prevented control flow from reaching `callback(null)` below. Also, Brandon and I had discussed that returning true/false from this function to indicate whether parsing succeeded or not was sensible API, and that is stripped by this change.

Comment (sandover) [inline index.js]: I don't understand the call-twice debate and don't have time to look into it at the moment.  I can look more deeply later. 

That side, though, I do agree with Eben -- I would like to preserve the semantics he mentions.

Comment (tomjakubowski) [inline index.js]: I also can't see how the callback would've been called twice in the old version. Could you elaborate?

I do find this whole function a little hard to follow, though, with the nested try/catch (wrapping the call of the callback in a try/catch seems especially odd to me).

File: index.js [hunk 2]
```diff
@@ -185,6 +185,7 @@ var oldest_idx = function oldest_idx(pool, callback) {
 //
 // Returns a bool to indicate whether parsing was successful
 var process_protein = function process_protein(protein, callback) {
+  var callbackCalled = false;
   try {
     yaml.safeLoadAll(protein, function(protein) {
```
Comment (orourkek) [inline index.js]: Most of the meat of this function was happening here, inside this `function`. The `return true` that was removed from below was returning from this function, not `process_protein`. Since the `callback(null)` was happening after this function but before the `return false` for `process_protein`, the callback was being called twice (once with `protein`, once with `null`).

I removed the returned `true`/`false` values only because no code seems to be actually using the return value, and due to the aforementioned bug, `process_protein` was always returning false.

Comment (ebeneliason) [inline index.js]: Ah, thanks Kevin. I completely missed the nested callback here. In that case, simply returning callbackCalled at the end would uphold the API we desired. It might be more clear to call it something like parsingSucceeded instead.

File: index.js (hunk 1 above)
Comment (orourkek) [inline index.js]: I updated the PR with a new commit that adds back the return values. See above for explanation of the double callback call.

Comment (ebeneliason) [inline index.js]: All looks good to me now, Kevin!

Comment (sandover) [inline index.js]: OK, merging.

**Comment by sandover** [comment]
Version 0.1.9 published to npm.
